src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,221 @@
+package duke;
+
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.TaskType;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to store and manage Task objects.
+ */
+public class TaskList {
+
+    List<Task> taskList;
+
+    /**
+     * Create an empty TaskList.
+     */
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    /**
+     * Creates a TaskList from the given list of representations.
+     *
+     * @param stringList List of string representations
+     */
+    public TaskList(List<String> stringList) {
+        taskList = new ArrayList<>();
+        stringList.forEach(this::parseLine);
+    }
+
+    /**
+     * Returns the list of Tasks.
+     *
+     * @return List of Tasks
+     */
+    public List<Task> getList() {
+        return taskList;
+    }
+
+    /**
+     * Returns the size of the list.
+     *
+     * @return size of list
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Creates a Task from the given representation and adds it to the list.
+     *
+     * @param line string repesentation of the Task
+     */
+    private void parseLine(String line) {
+        String[] lineSplit = line.split("\\|");
+        boolean done = lineSplit[2].equals("1");
+        if (lineSplit[0].equals("T")) {
+            addTask(lineSplit[1], TaskType.TODO, "", done);
+        } else if (lineSplit[0].equals("D")) {
+            addTask(lineSplit[1], TaskType.DEADLINE, lineSplit[3], done);
+        } else {
+            addTask(lineSplit[1], TaskType.EVENT, lineSplit[3], done);
+        }
+    }
+
+    /**
+     * Adds a Task to the list.
+     *
+     * @param task Task name
+     * @param type TaskType of the of Task to add
+     * @param ddl Task deadline
+     * @param done whether task has been marked done
+     */
+    private void addTask(String task, TaskType type, String ddl, boolean done) {
```

Comment:
Perhaps we can create new tasks to be added at the `AddCommand` class, and pass the task to `TaskList`, that way, it would only need to add it to the `taskList`?